gamestonk_terminal/portfolio/portfolio_optimization/optimizer_view.py

```diff
@@ -342,13 +342,20 @@ def display_ef(
     ef, rets, stds = optimizer_model.generate_random_portfolios(
         stocks, period, n_portfolios
     )
+    # The ef needs to be deep-copied to avoid error in plotting sharpe
+    ef2 = copy.deepcopy(ef)
```

Comment:
interesting, had never used this before!

Reply:
me neither.  I had to dive into the portfolio source code